src/api/app/models/cloud/upload_job.rb

```diff
@@ -4,32 +4,51 @@ class UploadJob
     include ActiveModel::Model
     extend Forwardable
 
-    attr_accessor :user_upload_job, :backend_upload_job
-    validate :validate_jobs
+    attr_accessor :user_upload_job, :backend_upload_job, :target_validator, :filename, :architecture, :target, :user
+    validate :validate_dependencies
+    validates :user, presence: true
+    validates :filename, presence: true, format: {
+      with: /\A.+(.raw.xz|.vhdfixed.xz)\z/, message: "'%{value}' is not a valid cloud image (needs to be a raw.xz or vhdfixed.xz file)"
```

Comment:
Nitpick: In such a case it's better to use ` .end_with? `

Reply:
AFAIK therefore I would need to write my own validates method, already investigated that!